### DISCUSSION TITLE
Icons, application and menu

### DISCUSSION THREAD

**Discussion by droc12345** [labwc/labwc#2076] [General]
I've gotten a decent implementation of client-list-combined-menu (minus icons) working pretty well.

I'm looking at what might be required for application icons (titlebar, client-list-*-menu) or just regular menus, like root-menu.

I would like feedback, suggestions, ideas, etc about not only implementing it but make something work for all cases
without having to reimplement every new feature. We do handle icons but only for buttons and most of the use cases
have nothing to do with buttons, so that might be a place to start, ie factor out the icon handling from the button use case.

Something like a function that takes an app-id and returns a buffer with an icon image in it, ready to be used.

Along with that how much do we want to get into chasing down an application icon, do we roll our own, maybe
something similar to how fuzzel gets theirs, or do we use something like #2068 ie a toolkit to get the icons and  
possibly put it into a buffer ready for use.

**Comment by Consolatis** [comment]
I think the current idea is to use https://gitlab.freedesktop.org/vyivel/libsfdo/ to look up icons.
We will also need some autoscaling cache to prevent a blurry mess on scales > 1.

**Comment by droc12345** [comment]
I looked at libsdfdo, not terribly hard to incorporate.

I know three ways we use or want to use icons right now.
1. inside the menu.xml file you can assign an icon (full path to it so no lookup needed)
2. for use with menus like client-list-combined menu (running apps)
3. titlebar use (app icon and menu buttons)

Probably want a lookup table app <-> icon name for quick reference. 
Done once at startup, or on a reconfigure event. 

The old obmenu.pl program (with mods for labwc/wayland vs openbox/X11) creates a nice static or pipemenu 
with the ability to produce icons, when run it finds all the icons it might need and puts them in XDG_CACHE_HOME dir.

**Reply by stefonarch** [reply]
>I know three ways we use or want to use icons right now.

IMO app switcher could show them as well.
